src/main/fc/fc_rc.c

```diff
@@ -98,37 +86,44 @@ static int16_t rcLookupThrottle(int32_t tmp)
 #define SETPOINT_RATE_LIMIT 1998.0f
 #define RC_RATE_INCREMENTAL 14.54f
 
-static void calculateSetpointRate(int axis)
+float applyBetaflightRates(int axis, float rcCommandf, const float rcCommandfAbs)
 {
-    uint8_t rcExpo;
-    float rcRate;
-    if (axis != YAW) {
-        rcExpo = currentControlRateProfile->rcExpo8;
-        rcRate = currentControlRateProfile->rcRate8 / 100.0f;
-    } else {
-        rcExpo = currentControlRateProfile->rcYawExpo8;
-        rcRate = currentControlRateProfile->rcYawRate8 / 100.0f;
+    if (currentControlRateProfile->rcExpo[axis]) {
+        const float expof = currentControlRateProfile->rcExpo[axis] / 100.0f;
+        rcCommandf = rcCommandf * power3(rcCommandfAbs) * expof + rcCommandf * (1 - expof);
     }
-    if (rcRate > 2.0f) {
-        rcRate += RC_RATE_INCREMENTAL * (rcRate - 2.0f);
+
+    float angleRate = 200.0f * currentControlRateProfile->rcRates[axis] * rcCommandf;
+    if (currentControlRateProfile->rates[axis]) {
+        const float rcSuperfactor = 1.0f / (constrainf(1.0f - (rcCommandfAbs * (currentControlRateProfile->rates[axis] / 100.0f)), 0.01f, 1.00f));
+        angleRate *= rcSuperfactor;
     }
 
+    return angleRate;
+}
+
+float applyRaceFlightRates(int axis, float rcCommandf, const float rcCommandfAbs)
+{
+    UNUSED(rcCommandfAbs);
+
+    // -1.0 to 1.0 ranged and curved
+    rcCommandf = ((1.0f + 0.01f * currentControlRateProfile->rcExpo[axis] * (rcCommandf * rcCommandf - 1.0f)) * rcCommandf);
+    // convert to -2000 to 2000 range using acro+ modifier
+    float angleRate = 10.0f * currentControlRateProfile->rcRates[axis] * rcCommandf;
```

Comment:
It may be worth coming up with a way to allow for 10x multiplier rates. e.g. for now, rates `370`, `380` are available (37*10); but it's common to want/have slightly more precision there, e.g. users report using 375deg/sec; I don't know of a good way to afford this without a uint16